assets/chezmoi.io/docs/reference/special-directories/index.md

```diff
@@ -1,6 +1,25 @@
 # Special directories
 
 All directories in the source state whose name begins with `.` are ignored by
-default, unless they are one of the special directories listed here.
-`.chezmoidata` and `.chezmoitemplates` are read before all other files so that
-they can be used in templates.
+default, unless they are one of the special directories listed here. All of
+these directories are optional and are evaluated in a specific order.
```

Comment:
This order is based on what I believe is correct; I have not verified it in the code. It would be good to get clarity.

Reply:
The order is defined here:

https://github.com/twpayne/chezmoi/blob/9e6d5369ef808968118560b6f2b6a09a8220375d/internal/chezmoi/system.go#L190-L199

Data and templates need to be evaluated before everything else, but for the rest it doesn't matter.